MaxText/max_utils.py

```diff
@@ -266,6 +270,59 @@ def initialize_jax_for_cpu():
   )
 
 
+def initialize_jax_for_tpu_with_emergency_checkpointing(raw_keys):
+  """Initialize JAX distributed runtime for TPUs when emergency checkpointing is used."""
+  ID_FILE = "id_file.txt"
+
+  local_id_file = epath.Path(raw_keys["local_checkpoint_directory"]) / ID_FILE
+  if local_id_file.exists():
+    max_logging.log("An ID file from a previous run exists, initializing JAX distributed runtime using the saved ID.")
+    process_id = int(local_id_file.read_text())
```

Comment:
This surprises me - there is a local file for each process? Does each process have a different "local_checkpoint_directory"?